src/calendar/calendar.jsx

```diff
@@ -25,9 +26,18 @@ export default class Calendar extends React.Component {
       React.PropTypes.object,
       React.PropTypes.func
     ]),
+    NoticeComponent: React.PropTypes.oneOfType([
+      React.PropTypes.object,
+      React.PropTypes.func
+    ]),
     activeMonth: React.PropTypes.instanceOf(Date),
     blockClassName: React.PropTypes.string,
     disableDaysOfWeek: React.PropTypes.bool,
+    disabledIntervals: React.PropTypes.arrayOf(React.PropTypes.shape({
+      // TODO: Proper prop type for all the dates
+      start: React.PropTypes.string.isRequired,
+      end: React.PropTypes.string.isRequired
```

Comment:
A date is either `number`, `string` (ISO 8601) or an instance of `Date`. Here and everywhere.